### ISSUE TITLE
update failed

### ISSUE THREAD

**Issue by nutVI** [microsoft/vscode#147593]
Issue Type: <b>Bug</b>

My vscode can't auto update and try to active update will show the error log:
> [2022-04-17 21:58:03.684] [main] [info] update#setState checking for updates
[2022-04-17 21:58:03.686] [main] [error] Error: net::ERR_INVALID_ARGUMENT
    at SimpleURLLoaderWrapper.<anonymous> (electron/js2c/browser_init.js:109:7068)
    at SimpleURLLoaderWrapper.emit (events.js:315:20)
[2022-04-17 21:58:03.686] [main] [info] update#setState idle

VS Code version: Code 1.60.0 (e7d7e9a9348e6a8cc8c03f877d39cb72e5dfb1ff, 2021-09-01T10:41:52.311Z)
OS version: Windows_NT x64 10.0.19043
Restricted Mode: No
Remote OS version: Linux x64 5.4.72-microsoft-standard-WSL2

**Comment by nutVI** [comment]
I just tried to overwrite the current latest stable vscode version. Successfully installed, but still fail to check update.

VS Code version: Code 1.66.2 (dfd34e8260c270da74b5c2d86d61aee4b6d56977, 2022-04-11T07:46:01.075Z)
OS version: Windows_NT x64 10.0.19043

**Comment by joaomoreno** [comment]
> [2022-04-17 21:58:03.686] [main] [error] Error: net::ERR_INVALID_ARGUMENT
> at SimpleURLLoaderWrapper. (electron/js2c/browser_init.js:109:7068)
> at SimpleURLLoaderWrapper.emit (events.js:315:20)

@deepak1556 Have you see this?

**Comment by nutVI** [comment]
@deepak1556 I did use proxy, and i tried to turning it off but it still fail.
[netlog.json.txt](https://github.com/microsoft/vscode/files/8531581/netlog.json.txt)

**Comment by deepak1556** [comment]
I don't see the error code `net::ERR_INVALID_ARGUMENT` in the attached log, does it still fail with the same error when you turn off the proxy ?

**Comment by nutVI** [comment]
The previous log is in the environment where the proxy is turned off.
And the new log and screenshot is with proxy enabled. And I am sure it still fail with the same error.
[netlog.json.txt](https://github.com/microsoft/vscode/files/8532523/netlog.json.txt)
![Snipaste_2022-04-21_23-43-00](https://user-images.githubusercontent.com/54772860/164500945-234596eb-3b87-40ff-bd42-a1c28cd2b88e.jpg)

**Comment by gadfly3173** [comment]
same issue, renderer log show this:
```log
[2022-04-24 01:56:26.669] [renderer1] [error] t.filter is not a function: TypeError: t.filter is not a function
    at L.getUserConfiguredCustomEditors (file:///C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.desktop.main.js:5620:884)
    at L.getAllCustomEditors (file:///C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.desktop.main.js:5621:50)
    at L.updateContexts (file:///C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.desktop.main.js:5624:543)
    at file:///C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.desktop.main.js:5619:941
    at c.fire (file:///C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.desktop.main.js:99:696)
    at F.doHandleActiveEditorChangeEvent (file:///C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.desktop.main.js:6763:290)
    at F.handleActiveEditorChange (file:///C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.desktop.main.js:6763:88)
    at file:///C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.desktop.main.js:6762:128
    at c.fire (file:///C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.desktop.main.js:99:696)
    at e.doSetGroupActive (file:///C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.desktop.main.js:7020:1001)
    at Object.fromJSON (file:///C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.desktop.main.js:7026:973)
    at f._deserializeNode (file:///C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.desktop.main.js:1012:535)
    at file:///C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.desktop.main.js:1012:378
    at Array.map (<anonymous>)
    at f._deserializeNode (file:///C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.desktop.main.js:1012:359)
    at f._deserialize (file:///C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.desktop.main.js:1012:264)
    at Function.deserialize (file:///C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.desktop.main.js:1012:198)
    at Function.deserialize (file:///C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.desktop.main.js:1021:752)
    at e.doCreateGridControlWithState (file:///C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.desktop.main.js:7026:858)
    at e.doCreateGridControlWithPreviousState (file:///C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.desktop.main.js:7026:446)
    at e.doCreateGridControl (file:///C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.desktop.main.js:7025:929)
    at e.createContentArea (file:///C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.desktop.main.js:7024:299)
    at e.create (file:///C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.desktop.main.js:3205:531)
    at file:///C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.desktop.main.js:7174:118
    at Array.forEach (<anonymous>)
    at L.renderWorkbench (file:///C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.desktop.main.js:7174:25)
    at file:///C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.desktop.main.js:7171:6
    at d.invokeFunction (file:///C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.desktop.main.js:1759:588)
    at L.startup (file:///C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.desktop.main.js:7170:633)
    at ee.open (file:///C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.desktop.main.js:8357:740)
    at async file:///C:/Program%20Files/Microsoft%20VS%20Code/resources/app/out/bootstrap-window.js:8:365
[2022-04-24 01:56:26.669] [renderer1] [error] Unexpected error getting editor overrides: TypeError: t.filter is not a function
[2022-04-24 01:56:26.669] [renderer1] [error] Unexpected error getting editor overrides: TypeError: t.filter is not a function
[2022-04-24 01:56:26.669] [renderer1] [error] t.filter is not a function: TypeError: t.filter is not a function
    at L.getUserConfiguredCustomEditors (file:///C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.desktop.main.js:5620:884)
    at L.getAllCustomEditors (file:///C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.desktop.main.js:5621:50)
    at L.updateContexts (file:///C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.desktop.main.js:5624:543)
    at file:///C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.desktop.main.js:5619:941
    at c.fire (file:///C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.desktop.main.js:99:696)
    at F.doHandleActiveEditorChangeEvent (file:///C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.desktop.main.js:6763:290)
    at F.onEditorsRestored (file:///C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.desktop.main.js:6762:849)
    at file:///C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.desktop.main.js:6762:52
[2022-04-24 01:56:26.669] [renderer1] [error] Unexpected error getting editor overrides: TypeError: t.filter is not a function
[2022-04-24 01:56:26.669] [renderer1] [error] Unexpected error getting editor overrides: TypeError: t.filter is not a function
[2022-04-24 01:56:26.669] [renderer1] [error] Unexpected error getting editor overrides: TypeError: t.filter is not a function
[2022-04-24 01:56:26.669] [renderer1] [error] Unexpected error getting editor overrides: TypeError: t.filter is not a function
[2022-04-24 01:56:26.669] [renderer1] [error] Unexpected error getting editor overrides: TypeError: t.filter is not a function
[2022-04-24 01:56:26.669] [renderer1] [error] Unexpected error getting editor overrides: TypeError: t.filter is not a function
[2022-04-24 01:56:26.792] [renderer1] [error] t.filter is not a function: TypeError: t.filter is not a function
    at L.getUserConfiguredCustomEditors (file:///C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.desktop.main.js:5620:884)
    at L.getAllCustomEditors (file:///C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.desktop.main.js:5621:50)
    at L.updateContexts (file:///C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.desktop.main.js:5624:543)
    at file:///C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.desktop.main.js:5619:726
    at c.fire (file:///C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.desktop.main.js:99:696)
    at d.update (file:///C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.desktop.main.js:4904:295)
    at h._handler (file:///C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.desktop.main.js:4903:786)
    at h._handle (file:///C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.desktop.main.js:4191:920)
    at h.acceptUsers (file:///C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.desktop.main.js:4191:827)
    at Function._handleExtensionPoint (file:///C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.desktop.main.js:5432:668)
    at F._doHandleExtensionPoints (file:///C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.desktop.main.js:5431:744)
    at F._startLocalExtensionHost (file:///C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.desktop.main.js:5443:309)
    at F._scanAndHandleExtensions (file:///C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.desktop.main.js:5442:435)
    at async F._initialize (file:///C:/Program Files/Microsoft VS Code/resources/app/out/vs/workbench/workbench.desktop.main.js:5427:584)
[2022-04-24 01:56:26.928] [renderer1] [error] net::ERR_INVALID_ARGUMENT: Error: net::ERR_INVALID_ARGUMENT
    at SimpleURLLoaderWrapper.<anonymous> (electron/js2c/browser_init.js:105:6395)
    at SimpleURLLoaderWrapper.emit (events.js:223:5)
[2022-04-24 01:56:47.271] [renderer1] [error] Network Error: Error: Network Error
    at e.exports (C:\Program Files\Microsoft VS Code\resources\app\node_modules.asar\tas-client-umd\lib\tas-client-umd.js:1:7511)
    at XMLHttpRequest.p.onerror (C:\Program Files\Microsoft VS Code\resources\app\node_modules.asar\tas-client-umd\lib\tas-client-umd.js:1:6478)
```

**Comment by deepak1556** [comment]
Is this issue present with latest insiders https://code.visualstudio.com/insiders ?

**Comment by nutVI** [comment]
> Is this issue present with latest insiders https://code.visualstudio.com/insiders ?

Oh, thanks for also following up on this question. Actually, I uninstalled vscode and deleted all environment variables about vscode. After reinstalling vscode, this problem (update failed) was solved. What can be determined now is that overwriting installation cannot solve this problem, and it is hard to say about others. 
Thanks again for your care about this issue. Finally, I would like to ask how to deal with this issue, whether it should be closed or not closed.

**Comment by nutVI** [comment]
> Is this issue present with latest insiders https://code.visualstudio.com/insiders ?

Oh, thanks for also following up on this question. Actually, I uninstalled vscode and deleted all environment variables about vscode. After reinstalling vscode, this problem (update failed) was solved. What can be determined now is that overwriting installation cannot solve this problem, and it is hard to say about others. Maybe the new version happens to fix that.
Thanks again for your care about this issue. Finally, I would like to ask how to deal with this issue, whether it should be "Close as completed", "Close as not planned" or other?
